model/task/generated_json_storage.go

```diff
@@ -82,7 +82,7 @@ func GeneratedJSONInsertWithS3Fallback(ctx context.Context, settings *evergreen.
 	}
 
 	grip.Info(message.Fields{
-		"message":            "successfully upserted generated JSON files into S3 as fallback due to document size limitation",
+		"message":            "successfully inserted generated JSON files into S3 as fallback due to document size limitation",
```

Comment:
I have a concern that it fails not because it hits the document limit, but because we run out of time passing everything over (because we hit the "unexpected end of JSON input" limit. Should we instead just case on if the files are greater than the document size limit rather than trying to insert, since we have the byte size already? Or check more error cases here?

Reply:
Awesome thanks!